numpy/linalg/linalg.py

```diff
@@ -2112,6 +2112,9 @@ def norm(x, ord=None, axis=None, keepdims=False):
     """
     x = asarray(x)
```

Comment:
That is `x = asarray(x) + 0.0`. it's a bit ugly, but there you go. I had a proposal to add a `mintype` keyword to `asarray` that would also have solved this problem.